app/scripts/lib/crypto/base64url.js

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/**
+ * Base64url encoding/decoding of SJCL bitArrays.
+ *
+ * SJCL natively supports standard base64 encoding but not the urlsafe
+ * variant, which is needed for JWTs.
+ */
+
+define([
+  'sjcl'
```

Comment:
I know it's early in the dev cycle still, though I wonder if sjcl can be loaded using requireOnDemand. IIRC, sjcl is pretty hefty.